examples/chip-tool/templates/commands.zapt

```diff
@@ -18,16 +21,52 @@
 #include <zap-generated/CHIPClientCallbacks.h>
 #include <zap-generated/CHIPClusters.h>
 
-// Value logging functions.  The non-generated ones depend on the
+#define SKIP_SPACE(iter)                                                                                                           \
+    do                                                                                                                             \
+    {                                                                                                                              \
+        while (isspace(*iter))                                                                                                     \
+        {                                                                                                                          \
+            iter++;                                                                                                                \
+        }                                                                                                                          \
+    } while (0)
+
+// Value logging and parsing functions. The non-generated ones depend on the
 // generated ones, so are placed here.
 namespace {
 
 {{#zcl_clusters}}
 {{#zcl_structs}}
 CHIP_ERROR LogValue(const char * label, size_t indent, {{zapTypeToDecodableClusterObjectType name ns=parent.name isArgument=true}} value);
+CHIP_ERROR ParseValue(const char *& iter, const char * end, {{zapTypeToEncodableClusterObjectType name ns=parent.name}} & value,
+                      std::vector<std::function<void(void)>> & freeFunctions);
 {{/zcl_structs}}
 {{/zcl_clusters}}
 
+CHIP_ERROR ParseValue(const char *& iter, const char * end, bool & value, std::vector<std::function<void(void)>> & freeFunctions);
```

Comment:
move these to some header file as well. They do not seem to depend on the template parameters.

Add unit tests for them.